### PR TITLE
Upgraded react, react-dom and isomorphic-style-loader.

### PR DESCRIPTION
**Pull request by mglace**
Upgraded react, react-dom and isomorphic-style-loader packages.  Upgraded react context references to new context API.

### CODE REVIEW

File: src/components/App.js [hunk 1]
```diff
@@ -44,20 +44,25 @@ const ContextType = {
...
-
   render() {
+    const { context, insertCss } = this.props;
+
     // NOTE: If you need to add or modify header, footer etc. of the app,
     // please do that inside the Layout component.
-    return React.Children.only(this.props.children);
+    return (
+      <StyleContext.Provider value={{ insertCss }}>
+        <ApplicationContext.Provider value={{ context }}>
```
Comment (piglovesyou) [inline src/components/App.js]: I think this causes subordinates to do `this.context.context.fetch`. What do you think here just passing `value={ context }`?

**Comment by piglovesyou** [comment]
@mglace Terribly sorry for the late response. Would you please fix the conflict?

File: src/components/App.js (hunk 1 above)
Comment (mglace) [reply src/components/App.js]: Fixed

**Comment by piglovesyou** [comment]
And the conflicts of `package.json` and `yarn.lock` too please if you don't mind. Or rather we can get the latest for them (`react`, `react-dom`, `isomorphic-style-loader`).

**Comment by mglace** [comment]
> And the conflicts of `package.json` and `yarn.lock` too please if you don't mind. Or rather we can get the latest for them (`react`, `react-dom`, `isomorphic-style-loader`).

I'm not exactly sure how to resolve the conflicts in the yarn.lock file.  Any advice on how to accomplish that?

**Comment by piglovesyou** [comment]
Usually just tracking `package.json` works. Why don't you run merge and take `--theirs` on `yarn.lock` and then run `yarn` again? Also I would suggest the latest version of these packages will be nice. (for instance isomorphic-style-loader is now 5.1.0)

**Comment by piglovesyou** [comment]
Please check out the build failure

**Comment by mglace** [comment]
I don't know jest very well but at first glance, it doesn't seem to be working with the new React context API.  I'll keep digging but if anyone else has any experience with this please let me know.

**Comment by mglace** [comment]
As I suspected, it looks like enzyme does not yet support statically setting contextType in React 16 (https://github.com/airbnb/enzyme/issues/1553).  I will look into alternative ways to test this functionality.

**Comment by mmarkelov** [comment]
@mglace you should update **babel-jest** and **jest** to pass your tests, after it you could remove **babel-core** also

**Comment by mglace** [comment]
@mmarkelov thanks for the suggestion but no luck i'm afraid.  I upgraded **babel-jest** and **jest** to the latest versions and I continue to receive the same error.

```
 FAIL  src/components/Layout/Layout.test.js
  Layout
    × renders children correctly (25ms)
  ● Layout › renders children correctly

    TypeError: Cannot read property 'apply' of undefined

      19 |   test('renders children correctly', () => {
      20 |     const wrapper = renderer
    > 21 |       .create(
         |        ^
      22 |         <App insertCss={() => {}} context={{ fetch: () => {}, pathname: '' }}>
      23 |           <Layout>
      24 |             <div className="child" />
```
I'll keep working on it.

**Comment by mmarkelov** [comment]
@mglace create PR to your branch - it is ok for me

**Comment by mmarkelov** [comment]
@mglace sorry man, I just forgot about lint( Could pls fix it? Or I can make PR

**Comment by piavgh** [comment]
Hi @mglace 

Thank you for this update. Is there a plan on when this PR will be merged into master?

**Comment by mmarkelov** [comment]
@mglace another problem. I just forgot that some packages are not incompatible with Node 6. **.travis.yml** should be changed. We just need to drop 6 - I can fix it =)

**Comment by piglovesyou** [comment]
Before dropping Node6, please

* Try finding way to keep supporting it
* Explain why it is necessary/inevitable
* Update all docs/related settings in the project

**Pull request closed without merging**